### ISSUE TITLE
Darglint reports error when an exception is raised with `.with_traceback(...)`

### ISSUE THREAD

**Issue by gschwaer** [terrencepreilly/darglint#181]
The following code generates two darglint errors:
```python
def traceback_test(self):
        """[summary]

        Raises:
            ValueError: [description]
        """
        try:
            print("")
        except BaseError as e:
            raise ValueError("...").with_traceback(e.__traceback__)
```

## Observed Behavior
```
test.py:traceback_test:42: DAR401: -r with_traceback
test.py:traceback_test:46: DAR402: +r ValueError
```
It seems darglint does not know of the [`with_traceback()` function](https://docs.python.org/3.7/library/exceptions.html#BaseException.with_traceback).

## Expected Behavior
In this case no error at all.

`with_traceback` should be interpreted as if it wasn't there because it returns the same exception object.

**Comment by leandro-lucarella-frequenz** [comment]
This extends to any `raise` that is not purely instantiating a class (or probably calling a function in general). For example:

```py
def test(self):
        """[summary]

        Raises:
            ValueError: [description]
        """
        try:
            print("")
        except ValueError as e:
            # Do something with e
            raise e
```

Will also produce a bogus:
```
test.py:traceback_test:42: DAR401: -r e
test.py:traceback_test:46: DAR402: +r ValueError
```

**Comment by leandro-lucarella-frequenz** [comment]
BTW, this can be workaround by using a `noqa`:

```py
def test(self):
        """[summary]

        Raises:
            ValueError: [description]

        # noqa: DAR401 e
        """
        try:
            print("")
        except ValueError as e:
            # Do something with e
            raise e
```

Or `with_traceback` instead of `e` in the original example.
